File: 10-linked-list/01-traverse-linked-list/01-traverse-linked-list.py
import sys
import os
sys.path.append(os.getcwd())
from PythonDataStructures.SinglyLinkedList.ListNode import ListNode

def traverse_linked_list(head):
  res = []
  while head:
    res.append(head.val)
    head = head.next
  return res


head = ListNode(
    1, ListNode(2, ListNode(3, ListNode(4))))
res = traverse_linked_list(head)
print(res)

File: 10-linked-list/01-traverse-linked-list/01-traverse-linked-list.ts
import { ListNode } from '../../TypeScriptDataStructure/SinglyLinkedList/ListNode'

const traverse_linked_list = (head: ListNode) => {
  const res = []
  while (head) {
    res.push(head.val)
    head = head.next
  }
  return res
}

const node = new ListNode(10, new ListNode(20))
console.log(traverse_linked_list(node))
